alf/bin/grid_search.py

```diff
@@ -258,10 +260,11 @@ def run(self):
         the configured space.
         """
 
-        # parsing gin configuration here to make all jobs have same copy
-        #   of base configuration (gin file may be changed occasionally)
-        gin_file = common.get_gin_file()
-        gin.parse_config_files_and_bindings(gin_file, FLAGS.gin_param)
+        # parsing gin/alf configuration here to make all jobs have same copy
+        #   of base configuration (conf file may be changed occasionally)
+        conf_file = common.get_conf_file()
+        # Do not init env seed here because this will init cuda in the main proc
+        common.parse_conf_file(conf_file, get_env=False)
```

Comment:
Some config need env in order to correctly run (e.g. get obervation_spec).

Reply:
I just moved the last step of the original parse_conf() (get_env()) to worker process below. The order is not changed.

Reply:
I see what you mean. In our new alf conf there are places that directly call get_env() in the conf file. Unfortunately this will init cuda device in the main process. I think there might be some way to delay evaluating any env spec (not evaluating them until actually using them).

Reply:
Why does the conf file needs to be parsed in the main process?

Reply:
If there are multiple jobs in the queue, some jobs will get launched after hours. During that period of time, the conf file might have been changed on disk.

Reply:
I can add an option of "init_random_seed" when calling `get_env()`. The calls in conf files will not use this option.